### PR TITLE
WIP: BenchmarkResult submission: re-work validation

### PR DESCRIPTION
**Pull request by jgehrcke**
This touches on a number of tickets and topics. This is work in progress (tests are not yet passing, and I still would like to do a bit of code cleanup), and I think I'd like to split this in to multiple commits / patches after all. But I wanted to get this up for an initial round of feedback/reviews before the weekend. Would appreciate some eyes on this, thanks!

### CODE REVIEW

File: conbench/entities/benchmark_result.py [hunk 1]
```diff
@@ -108,174 +158,259 @@ class BenchmarkResult(Base, EntityMixin):
...
+
+        run = Run.first(id=userres["run_id"])
+        if run is not None:
+            if userres.get("github") is not None:
+                chrun = run.commit.hash
+                chresult = userres["github"]["commit"]
+                if chrun != chresult:
+                    raise BenchmarkResultValidationError(
+                        f"result refers to commit hash {chresult}, but Run {run.id} "
+                        f"refers to commit hash {chrun}."
```
Comment (jgehrcke) [inline conbench/entities/benchmark_result.py:194]: this is about https://github.com/conbench/conbench/issues/932

File: conbench/entities/benchmark_result.py [hunk 2]
```diff
@@ -108,174 +158,259 @@ class BenchmarkResult(Base, EntityMixin):
...
+
+            if stats_usergiven["iterations"] > 1:
+                # Note(JP): I think that this here is precisely the one way to
+                # report the result of a microbenchmark with cross-repetition
+                # stats: data: length 1 (the duration of running many
+                # repetitions), and iterations: a number larger than 1, while
+                # mean, max, are now set (then these can be assumed to be
+                # derived from _within_ the microbenchmark. Note that this is a
+                # very special condition). The mean time for a single
+                # repetition now is sample/iterations.
```
Comment (alistaire47) [inline conbench/entities/benchmark_result.py:306]: The issue with this is the frameworks that do this (googlebench, folly) can also run multiple repetitions, which would have to get reported like multisamples, so `iterations` etc would mean two different things for those cases. Maybe eventually we could accept per-iteration (read: repetition) stats?

Comment (jgehrcke) [reply conbench/entities/benchmark_result.py:306]: Thanks! New PR with this code section is here #1119, and I've linked your comment from here: https://github.com/conbench/conbench/pull/1119#discussion_r1167300853

**Comment by jgehrcke** [comment]
This was superseded by the PRs linked above. Closing.

**Pull request closed without merging**